cloudwatch_monitoring_role/main.tf

```diff
@@ -0,0 +1,21 @@
+data "aws_iam_policy_document" "cloudwatch_monitoring_policy" {
+  statement {
+    sid     = "1"
+    actions = ["cloudwatch:PutMetricData", "cloudwatch:GetMetricStatistics", "cloudwatch:ListMetrics", "ec2:DescribeTags"]
+    effect  = "Allow"
+
+   resources = ["*"]
+  }
+}
+
+resource "aws_iam_policy" "cloudwatch_monitoring_policy" {
```

Comment:
I think you forgot your outputs

Reply:
Like what? I don't think we need any.. do I miss something?

Reply:
ah you attach inside the module.

Maybe output the policy and make the attachment optional? That way you can add additional policies to an instance instead of only the cloudwatch one.

Reply:
nvm, you are attaching to the role, not attaching the role to the instance, my bad